### PR TITLE
Added ability to set default min/magnification filtering 

### PR DESCRIPTION
**Pull request by Super-Genius**
for 2D Textures and Font atlas.

The default min/magnification settings for rendering textures was GL_LINEAR, which blurred textures.  This allows the user to set the default filtering when creating textures.

### CODE REVIEW

**Comment by pandamicro** [comment]
@xiaofeng11 @liamcindy Please review this PR, @WenhaiLin please also confirm

**Comment by xiaofeng11** [comment]
Logic of these code is OK, but I'm not familiar with OpenGL programming, @liamcindy @WenhaiLin please confirm does these setting change will change the render effect of follow-up operations

File: cocos/2d/CCFontFreeType.cpp [hunk 1]
```diff
@@ -38,6 +39,8 @@ FT_Library FontFreeType::_FTlibrary;
 bool       FontFreeType::_FTInitialized = false;
 const int  FontFreeType::DistanceMapSpread = 3;
 
+int FontFreeType::_defaultFontDPI;
```
Comment (WenhaiLin) [inline cocos/2d/CCFontFreeType.cpp]: Please specify the initial value of `FontFreeType::_defaultFontDPI`.

**Comment by minggo** [comment]
This PR includes so many unneeded commits. So i will close it.

**Comment by minggo** [comment]
@klhurley Could you please send a new PR?

**Pull request closed without merging**